client/modules/IDE/components/PreviewFrame.jsx

```diff
@@ -81,10 +82,38 @@ class PreviewFrame extends React.Component {
 
   handleConsoleEvent(messageEvent) {
     if (Array.isArray(messageEvent.data)) {
-      const decodedMessages = messageEvent.data.map(message => Object.assign(Decode(message.log), { source: message.source }));
-
+      const decodedMessages = messageEvent.data.map(message => Object.assign(
+        Decode(message.log),
+        { source: message.source }
+      ));
       decodedMessages.every((message, index, arr) => {
-        const { data: args } = message;
+        const { data: args, source } = message;
+        if (source === 'console') {
+          let consoleInfo = '';
+          const consoleBuffer = [];
+          const LOGWAIT = 100;
+          Hook(window.console, (log) => {
```

Comment:
is this Hook/Unhook/postMessage part necessary? would it instead make sense to just eval the input from the ConsoleInput, and then dispatch the action `dispatchConsoleEvent`? rather than faking it like this?

Reply:
These tricky code is to ensure that when the user enters the console method in the console, it can be captured correctly. Because the `eval` is running in the parent window, we can not catch the console info in the console when using previous `Hook` function(running in iframe window).

Reply:
I think it's necessary to use `Hook/Unhook` here, but maybe we can find a more elegant way.